latch/registry/table.py

```diff
@@ -0,0 +1,275 @@
+import asyncio
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Optional, Type, TypedDict
```

Comment:
`typing.Iterator` was deprecated in 3.9 fyi, tho I think we support 3.8 so don't think you can get rid of it